torch/_inductor/scheduler.py

```diff
@@ -2166,7 +2166,16 @@ def can_eliminate_user(user: NodeUser) -> bool:
                 else:
                     active_buffers = True
 
-            can_eliminate = not node.has_side_effects() and not active_buffers
+            # Node's read_writes doesn't contain unbacked symbols, so if the output buffer defines an unbacked symint,
```

Comment:
It's larger blast radius to do it in `get_read_write` and I think it would take some more discussions to come up with a proper fix. For now, can you try add below to ir.Conditional? I can stamp for you. 
 ```python
   # TODO: handle 
    def get_unbacked_symbol_uses(self) -> OrderedSet[Symbol]:
        out = OrderedSet()
        for t in [self.predicate] + self.operands:
            if isinstance(t, ShapeAsConstantBuffer):
                out = out.union(
                    free_unbacked_symbols(t.expr)
                )
        return out
    ```